tests/test_profiler.py

```diff
@@ -1,55 +1,96 @@
 import time
-
 import numpy as np
+import pytest
 
 from pytorch_lightning.profiler import Profiler, AdvancedProfiler
 
+PROFILER_OVERHEAD_MAX_TOLERANCE = 0.0001
+
+
+@pytest.fixture
+def simple_profiler():
+    profiler = Profiler()
+    return profiler
 
-def test_simple_profiler():
-    p = Profiler()
 
-    with p.profile("a"):
-        time.sleep(3)
+@pytest.fixture
+def advanced_profiler():
+    profiler = AdvancedProfiler()
+    return profiler
 
-    with p.profile("a"):
-        time.sleep(1)
 
-    with p.profile("b"):
-        time.sleep(2)
+@pytest.mark.parametrize("action,expected", [("a", [3, 1]), ("b", [2]), ("c", [1])])
+def test_simple_profiler_durations(simple_profiler, action, expected):
+    """
+    ensure the reported durations are reasonably accurate
```

Comment:
it took a mi sec to understand the idea behind, that is looping with the same instance... maybe it would be nice to write it in doc, in someone in future would need to change/update it...

Reply:
i think it might be a good idea to update the `test/README.md` with testing philosophy and guidelines. introduce the usage of fixtures/parameterized tests, specify the desired level of documentation, etc. 

what do you think?

Reply:
good point, just not README but CONTRIBUTION on docs - in another issue/PR?
still, please write the two lines about the holding instance for multiple runs (i assume it is a bit special case...) Thx :]